### PR TITLE
Feature/dreams change color after being read  patch 0

### PR DESCRIPTION
**Pull request by data-doge**
@will-sklenars 

ey dude, small patch to your PR here -- plz merge if you're into it

changes summarized in the commit messages

### CODE REVIEW

**Comment by data-doge** [comment]
sick -- after creating this partner API PR https://github.com/enspiral-cherubi/dREAM-cACHER/pull/5, i added a bunch more to this branch to get the two to talk to eachother.

in summary: 
- refactored dream fetching fxns in controller
- when dreams are fetched, the `current_user`'s `views` are also fetched and used to set a `viewed` boolean on the client-side dream models stored in `dreamData`

File: js/app/environment.js [hunk 1]
```diff
@@ -198,4 +199,17 @@ environment.render = function () {
   })
 }
 
+environment.markDreamAsViewed = function (dreamIndex) {
+  var clonedGeom = this.dreamsMesh.geometry.clone()
+  this.removeObjectFromScene(this.dreamsMesh)
+  var faceIndices = this.pickingData[dreamIndex].faceIndices
+  range(faceIndices.low, faceIndices.hi).forEach(function (faceIndex) {
```
Comment (will-sklenars) [inline js/app/environment.js:206]: Niiice! The range thing. That is so much cleaner.

File: js/app/environment.js [hunk 2]
```diff
@@ -198,4 +199,17 @@ environment.render = function () {
...
+    clonedGeom.faces[faceIndex].materialIndex = 1
```
Comment (will-sklenars) [inline js/app/environment.js:207]: Yeah, It is interesting how we have to do this. Like, I feel like we should be able to just update the material in the same way that you can change the opacity of a material in run time and it updates. I couldn't get it to work like that though. I feel like maybe this is a quirk of the multiple material biz. I though that just updating the geometry and then creating a new mesh with a new material and the old geom would work, but that doesn't register the change in material indexes. It must compile that information when the geometry is instantiated, or something of the like.

File: js/app/model.js [hunk 3]
```diff
@@ -89,22 +72,50 @@ var dreamsModel =  {
     })
   },
 
-  getDreamsForTag: function (tag) {
+  // private
+
+  pathForFetchType: function (fetchType) {
```
Comment (will-sklenars) [inline js/app/model.js:77]: Nice!!

File: js/app/model.js [hunk 4]
```diff
@@ -89,22 +72,50 @@ var dreamsModel =  {
...
+    if (typeof fetchType === 'object' && fetchType.tag) {
+      return '/tag/dreams?tag=' + fetchType.tag
+    } else if (fetchType === 'forUser') {
+      return '/user/dreams'
+    } else {
+      return '/dreams'
+    }
+  },
+
+  fetchViews: function () {
```
Comment (will-sklenars) [inline js/app/model.js:87]: Don't fully understand what is going on here, But I will look at it more closely tomorrow. Gotta do contracting tonight.

File: js/services/apply-vertex-colors-to-geometry.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+var THREE = require('three')
+
+module.exports = function (geometry, color) {
```
Comment (will-sklenars) [inline js/services/apply-vertex-colors-to-geometry.js:3]: good idea to split that out

**Comment by will-sklenars** [comment]
Nice man!!!!!! This is great! I'm going to have a look at this in more detail tomorrow because I wana learn from some of these refactorings that you have done, but I am happy to merge.

**Pull request merged**